wgengine/router/router_linux.go

```diff
@@ -932,3 +841,72 @@ func normalizeCIDR(cidr netaddr.IPPrefix) string {
 	nip := ncidr.IP.Mask(ncidr.Mask)
 	return fmt.Sprintf("%s/%d", nip, cidr.Bits)
 }
+
+// upDNS updates the system DNS configuration to the given one.
+func (r *linuxRouter) upDNS(config DNSConfig) error {
+	if len(config.Nameservers) == 0 {
+		return r.downDNS()
+	}
+
+	switch r.dnsMode {
+	case dnsResolved:
+		if err := dnsResolvedUp(config); err != nil {
+			return fmt.Errorf("resolved: %w", err)
+		}
+	case dnsResolvconf:
+		if err := dnsResolvconfUp(config, r.tunname); err != nil {
+			return fmt.Errorf("resolvconf: %w", err)
+		}
+	case dnsNetworkManager:
+		if err := dnsNetworkManagerUp(config, r.tunname); err != nil {
+			return fmt.Errorf("network manager: %w", err)
+		}
+	case dnsDirect:
+		if err := dnsDirectUp(config); err != nil {
+			return fmt.Errorf("direct: %w", err)
+		}
+		exec.Command("service", "systemd-resolved", "restart").Run()
```

Comment:
This is one of those things I assumed was necessary in some edge case because it was there before. After some testing it doesn't seem to make a difference, so I'll remove it.